File: examples/equos-python-integration/src/python_integration/create_voice.py
import sys

from equos.client.errors import UnexpectedStatus
from equos.models import CreateEquosVoiceRequest, CreateEquosVoiceRequestIdentity

from python_integration.client import get_client


def create_voice() -> None:
    client = get_client()
    identity = CreateEquosVoiceRequestIdentity.CHARON

    try:
        voice = client.voices.create(
            CreateEquosVoiceRequest(
                identity=identity,
                instructions="Warm, friendly, conversational tone.",
            )
        )
        print("Voice created:", voice.id)
    except UnexpectedStatus as err:
        if err.status_code != 409:
            raise
        existing = client.voices.list(take=50)
        match = next(
            (v for v in existing.voices if v.identity.value == identity.value),
            None,
        )
        print(
            f'A voice with identity "{identity.value}" already exists for this organization.',
            file=sys.stderr,
        )
        if match is not None:
            print(f"→ Reuse id: {match.id}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    create_voice()

File: examples/equos-nodejs-integration/src/create-voice.ts
import {
  CreateEquosVoiceRequestIdentityEnum,
  ResponseError,
} from "@equos/node-sdk";
import { client } from "./client.js";

// `identity` picks a voice from the Equos catalog. Full list in the SDK enum.
try {
  const voice = await client.voices.createVoice({
    createEquosVoiceRequest: {
      identity: CreateEquosVoiceRequestIdentityEnum.Charon,
      instructions: "Warm, friendly, conversational tone.",
    },
  });
  console.log("Voice created:", voice.id);
  console.log(voice);
} catch (err) {
  if (err instanceof ResponseError && err.response.status === 409) {
    const existing = await client.voices.listVoices({ take: 50 });
    const match = existing.voices.find(
      (v) => v.identity === CreateEquosVoiceRequestIdentityEnum.Charon
    );
    console.error(
      `A voice with identity "${CreateEquosVoiceRequestIdentityEnum.Charon}" already exists for this organization.`
    );
    if (match) console.error(`→ Reuse id: ${match.id}`);
    process.exit(1);
  }
  throw err;
}
